### PR TITLE
feat(.github/actions): add `create-release-v1` action

### PR DESCRIPTION
**Pull request by Zidious**
This PR adds a new composite action for creating a release. It will create a PR merging `release` into the provided `base` branch and within the body of the PR will be the changelog of the new additions. Additionally, there will be two optional inputs `migrations-dir` and `env-file-path`. For `migrations-dir` if supplied will check the diff and create a permalink to the migration file and put into the body of the release PR. Similarly, if `env-file-path` is supplied will extract the new env vars (with the values redacted) and also place them into the PR body. 

Example without the inputs: https://github.com/dequelabs/zidious-testing/pull/118 -- PR: https://github.com/dequelabs/zidious-testing/pull/118
Example with inputs: https://github.com/dequelabs/zidious-testing/pull/120 -- PR: https://github.com/dequelabs/zidious-testing/pull/120

Closes: https://github.com/dequelabs/axe-api-team-public/issues/89

### CODE REVIEW

File: .github/actions/create-release-v1/action.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    description: The path to the `.env` file e.g. src/service/sample.env
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        # Fetch all history
+        fetch-depth: 0
```
Comment (straker) [inline .github/actions/create-release-v1/action.yml:23]: Should checkout the release branch here

```suggestion
        fetch-depth: 0
        ref: 'release'
```

File: .github/actions/create-release-v1/action.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        echo "A release is already in progress. Cancelling this release."
+        gh run cancel ${{ github.run_id }}
+        gh run watch ${{ github.run_id }}
+
+    - name: Get release version
+      id: get_release_version
+      shell: bash
+      run: |
+        git checkout release
+
```
Comment (straker) [inline .github/actions/create-release-v1/action.yml:44]: Don't need to checkout here if actions/checkout did it

```suggestion
```

File: .github/actions/create-release-v1/action.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      run: |
+        echo "## Changelog Additions:" >> /tmp/release-notes.txt
+        echo "" >> /tmp/release-notes.txt
+
+        new_changelog_entries=$(git show origin/${{ inputs.base }}...origin/release --no-color --no-prefix --output-indicator-new=! CHANGELOG.md | egrep '^!' | awk -F'^[!]' '{print $2}' | sed -e 's/\n/$0A/g')
+
+        echo "$new_changelog_entries" >> /tmp/release-notes.txt
+        echo "" >> /tmp/release-notes.txt
+
+    - name: Find new migration files
```
Comment (straker) [inline .github/actions/create-release-v1/action.yml:65]: The migration step skips in the bash script while the env file skips using GHA `if` clause. Should make it consistent.

**Review by straker — DISMISSED** [review]

**Comment by Zidious** [comment]
With env vars and migrations: https://github.com/dequelabs/zidious-testing/pull/121 // https://github.com/dequelabs/zidious-testing/actions/runs/7462504271/job/20304951364
Without: https://github.com/dequelabs/zidious-testing/pull/122 // https://github.com/dequelabs/zidious-testing/actions/runs/7462528208/job/20305031902

**Review by straker — APPROVED** [review]

**Pull request merged**